tinygrad/tensor.py

```diff
@@ -182,7 +182,8 @@ def randn(*shape, dtype:Optional[DType]=None, **kwargs) -> Tensor:
   def normal(*shape, mean=0.0, std=1.0, **kwargs) -> Tensor: return (std * Tensor.randn(*shape, **kwargs)) + mean
 
   @staticmethod
-  def uniform(*shape, low=-1.0, high=1.0, **kwargs) -> Tensor: return ((high-low) * Tensor.rand(*shape, **kwargs)) + low
+  def uniform(*shape, low=-1.0, high=1.0, dtype:Optional[DType]=None, **kwargs) -> Tensor:
+    return ((high-low) * Tensor.rand(*shape, **kwargs)).cast(Tensor.default_type if dtype is None else dtype) + low
```

Comment:
is dtype not getting passed via kwargs?

Reply:
Ah, yes, looking into this now

Reply:
I believe your approach is kinda true, if you give dtype with kwargs Tensor.rand is converted to 0 or 1 implicitly.

Reply:
I think my latest commit (pruning dtype from kwargs) addresses this issue?